runtime/doc/lua.txt

```diff
@@ -1536,6 +1536,17 @@ deepcopy({orig})                                              *vim.deepcopy()*
     Return: ~
         (table) Table of copied keys and (nested) values.
 
+defaulttable({create})                                    *vim.defaulttable()*
+    Creates a table whose values are automatically created when missing.
+
+    They are kind of like defaultdicts in python
+
+    Parameters: ~
+        {create}  The function called to create a missing value
```

Comment:
Can we allow this to be `nil` which just means `function() return {} end`?

Reply:
Sounds reasonable.

Reply:
Maybe it could be even better to do `vim.defaulttable` as the default `create` function.
This way one can do as you suggested.

Reply:
Done